src/utils.py

```diff
@@ -95,34 +78,38 @@ def is_ipv4(ip: str) -> bool:
         return False
 
 
-# Service functions
 def _systemctl(action: str, service_name: str) -> None:
-    subprocess.run(["systemctl", action, service_name]).check_returncode()
+    shell(f"systemctl {action} {service_name}")
 
 
 def service_start(service_name: str) -> None:
     """Starts a given service."""
     _systemctl("start", service_name)
+    logger.info(f"Service {service_name} started")
```

Comment:
f-strings are discouraged with logging calls, because they are always interpreted. If the log level is disabled, the f-string will still be interpreted. 
```suggestion
    logger.info("Service %s started", (service_name))
```